src/EGraphs/egraph.jl

```diff
@@ -42,7 +42,7 @@ they represent. The [`EGraph`](@ref) itself comes with pretty printing for human
 struct EClass{D}
   id::Id
   nodes::Vector{VecExpr}
```

Comment:
Why not also have this be a vector of `Id`, if we now store `VecExpr`s in `nodes`?

Reply:
Yes, this could be better. I tested this already, and reverted it again because it introduces an awkward indirection of `g.nodes[class.nodes[i]]` in several places, most notably the ematching code. Local test showed no performance improvements of only storing the ids here. I'm not yet decided.

My current approach was to store the same nodes objects (VecExpr) in the `nodes` vectors of eclasses and the egraph. As far as I understand, we do not need to keep the original (uncanonicalized) enodes.